src/core/withExperiment.js

```diff
@@ -0,0 +1,114 @@
+/* @flow */
+import * as React from 'react';
+import invariant from 'invariant';
+import cookie from 'react-cookie';
+
+import tracking from 'core/tracking';
+
+type Props = {|
+  AName: string,
+  BName: string,
+  nameId: string,
+|};
+
+type InternalProps = {|
+  ...Props,
+  _cookie: typeof cookie,
+  _tracking: typeof tracking,
+  randomizer: () => number,
+  WrappedComponent: Function,
+|};
+
+export const withExperiment = ({ nameId, AName, BName }: Props) => (
+  WrappedComponent: Function,
+) => {
+  class WithExperiment extends React.Component<InternalProps> {
+    abTestCookie: string | void;
+
+    static defaultProps = {
+      _cookie: cookie,
+      _tracking: tracking,
+      AName,
+      BName,
+      nameId,
+      randomizer: Math.random,
+    };
+
+    componentWillMount() {
+      const {
+        _cookie,
+        _tracking,
+        AName: ANameVariant,
+        BName: BNameVariant,
+        nameId: abNameId,
+        randomizer,
+      } = this.props;
+
+      invariant(ANameVariant, 'AName is required');
+      invariant(BNameVariant, 'BName is required');
+      invariant(abNameId, 'nameId is required');
+
+      this.abTestCookie = _cookie.load(`AB_${abNameId}_COOKIE`);
+
+      if (this.abTestCookie === undefined) {
+        this.abTestCookie =
+          randomizer() >= 0.5
```

Comment:
this should probably be a prop of `withExperiment`, so that it is not a HOC that does 50-50, but it can probably be done in a follow-up issue.

Reply:
ok, I was wondering about this one too but figured 50-50 was common with these types of test

Reply:
no need to change that right now

Reply:
okie..